### PR TITLE
CASSANDRA-17575 3.11: Fix compaction token range bounds

### PR DESCRIPTION
**Pull request by adelapena**

### CODE REVIEW

File: src/java/org/apache/cassandra/db/compaction/CompactionManager.java [hunk 1]
```diff
@@ -812,11 +812,20 @@ protected void runMayThrow()
...
+     *
+     * @param cfStore The column family store to be compacted.
+     * @param ranges The token ranges to be compacted, interpreted as closed intervals.
+     */
     public void forceCompactionForTokenRange(ColumnFamilyStore cfStore, Collection<Range<Token>> ranges)
     {
         Callable<Collection<AbstractCompactionTask>> taskCreator = () -> {
             Collection<SSTableReader> sstables = sstablesInBounds(cfStore, ranges);
-            if (sstables == null || sstables.isEmpty())
+            if (sstables.isEmpty())
```
Comment (ekaterinadimitrova2) [inline src/java/org/apache/cassandra/db/compaction/CompactionManager.java]: This change is unrelated to the patch and my personal preference is to stick to the main patch in older versions

File: src/java/org/apache/cassandra/service/StorageServiceMBean.java [hunk 2]
```diff
@@ -265,8 +265,14 @@ public interface StorageServiceMBean extends NotificationEmitter
...
     public int relocateSSTables(int jobs, String keyspace, String ... cfnames) throws IOException, ExecutionException, InterruptedException;
+
     /**
-     * Forces major compaction of specified token range in a single keyspace
+     * Forces major compaction of specified token range in a single keyspace.
+     *
+     * @param keyspaceName the name of the keyspace to be compacted
+     * @param startToken the token at which the compaction range starts (inclusive)
+     * @param endToken the token at which compaction range ends (inclusive)
+     * @param tableNames the names of the tables to be compacted
```
Comment (ekaterinadimitrova2) [inline src/java/org/apache/cassandra/service/StorageServiceMBean.java:275]: Do we want also to add java doc to `forceKeyspaceCompactionForTokenRange` in `NodeProbe`, for completeness?

**Review by ekaterinadimitrova2 — APPROVED** [review]
Overall LGTM, left just 2 suggestions, thanks

File: src/java/org/apache/cassandra/db/compaction/CompactionManager.java (hunk 1 above)
Comment (adelapena) [reply src/java/org/apache/cassandra/db/compaction/CompactionManager.java]: Fine with me. The unneeded null check isn't on trunk nor 4.1, so it will remain on 3.11 and 4.0.

File: src/java/org/apache/cassandra/service/StorageServiceMBean.java (hunk 2 above)
Comment (adelapena) [reply src/java/org/apache/cassandra/service/StorageServiceMBean.java:275]: Makes sense, added.
